src/Not/Not/Authentication/NAuthenticationSettings.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.Extensions.Configuration;
+using Not.Authentication.User;
+
+namespace Not.Authentication;
+
+public class NAuthenticationSettings : IUserDeserializer
+{
+    public List<NUser> Users { get; set; } = [];
+
+    public Dictionary<string, NUser> GetAllowedUsers(IConfiguration configuration)
+    {
+        Users = configuration.GetSection("Auth:Users").Get<List<NUser>>() ?? [];
```

Comment:
While this works, it's not very convinient and callers will have to supply the settings with `IConfiguration` instances every time. Instead the settings class should be registered as a singleton object, which is bound to the relevat IConfiguration section during service configuration: [details here](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-10.0#bind-hierarchical-configuration-data-using-the-options-pattern)